CC-14562: Add categories to discount query parameters. (#665)
spryker/spryker-docs: docs/scos/dev/feature-integration-guides/202200.0/category-management-promotions-and-discounts-feature-integration.md

---
title: Category Management + promotions & discounts feature integration

> description: This guide provides step-by-step instructions on integrating Category Management + Promotions & Discounts feature into a Spryker-based project.

Reviewer: ```suggestion
description: This guide provides step-by-step instructions on integrating into a Spryker-based project the new feature: Addition of a "Category" parameter to calculation and conditions queries in discounts.
```

> description: his guide provides step-by-step instructions on integrating into a Spryker-based project the new feature: Addition of a "Category" parameter to calculation and conditions queries in discounts.

Reviewer 2: ```suggestion
description: This guide provides step-by-step instructions on integrating into a Spryker-based project the new feature: Addition of a "Category" parameter to calculation and conditions queries in discounts.
```

template: feature-integration-guide-template
---

## Install feature core

Follow the steps below to install the feature core.

### Prerequisites

To start feature integration, overview, and install the necessary features:

| NAME                   | VERSION |
|------------------------| --- |
| Promotions & Discounts | {{page.version}} |
| Category Management    | {{page.version}} |
| Spryker Core           | {{page.version}} |

> ### 1) Set up behavior

Reviewer 2: transfers section is missing

Reviewer 2: the same for frontend section 
CSS was adjusted

use Spryker\Zed\CategoryDiscountConnector\Communication\Plugin\Discount\CategoryDiscountableItemCollectorPlugin;
use Spryker\Zed\Discount\DiscountDependencyProvider as SprykerDiscountDependencyProvider;

class DiscountDependencyProvider extends SprykerDiscountDependencyProvider
{
    /**
     * @return array<\Spryker\Zed\DiscountExtension\Dependency\Plugin\DecisionRulePluginInterface>
     */
    protected function getDecisionRulePlugins(): array
    {
        return array_merge(parent::getDecisionRulePlugins(), [
            new CategoryDecisionRulePlugin(),
        ]);
    }

    /**
     * @return array<\Spryker\Zed\DiscountExtension\Dependency\Plugin\DiscountableItemCollectorPluginInterface>
     */
    protected function getCollectorPlugins(): array
    {
        return array_merge(parent::getCollectorPlugins(), [
            new CategoryDiscountableItemCollectorPlugin(),
        ]);
    }
}
```

{% info_block warningBox "Verification" %}

Ensure that the plugins work correctly:

1. [Create a discount](/docs/scos/user/back-office-user-guides/{{page.version}}/merchandising/discount/creating-cart-rules.html) and define its condition as a query string with a *category* field.

> 2. Add a product with the defined category to cart.

Reviewer: ```suggestion
2. Add a product assigned to the defined category to cart.
```

---
title: Category Management + Promotions & Discounts feature integration
description: Add the "category" parameter to calculation and conditions queries in the Promotions & Discounts feature.
last_updated: Jan 11, 2022
template: feature-integration-guide-template
---

This document describes how to add the `category` parameter to calculation and conditions queries in the [Promotions & Discounts](/docs/scos/user/features/{{page.version}}/promotions-discounts-feature-overview.html) feature.

## Install feature core

Follow the steps below to install the feature core.

### Prerequisites

To start feature integration, integrate the required features:

| NAME | VERSION | INTEGRATION GUIDE |
| --- | --- | --- |
| Promotions & Discounts | {{page.version}} | [Promotions & Discounts feature integration](/docs/scos/dev/feature-integration-guides/{{page.version}}/promotions-and-discounts-feature-integration.html) |
| Category Management | {{page.version}} | [Category Management feature integration](/docs/scos/dev/feature-integration-guides/{{page.version}}/category-management-feature-integration.html) |

> | Spryker Core | {{page.version}} | [Spryker Сore feature integration](/docs/scos/dev/feature-integration-guides/{{page.version}}/spryker-core-feature-integration.html) |

Reviewer 3: @dmiseev check the links I added to the feature integration guides and say if they are correct or I should replace them.
[Promotions & Discounts feature integration](https://docs.spryker.com/docs/scos/dev/feature-integration-guides/202108.0/promotions-and-discounts-feature-integration.html)
[Category Management feature integration](https://docs.spryker.com/docs/scos/dev/feature-integration-guides/202108.0/category-management-feature-integration.html)
[Spryker Сore feature integration](https://docs.spryker.com/docs/scos/dev/feature-integration-guides/202108.0/spryker-core-feature-integration.html)

Author: They are correct
